async-await.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+import {} from 'babel/polyfill';
+import fs from 'fs';
+import path from 'path';
+
+const dir = process.argv[2] || 'input';
+
+// fs module polyfill: this will be unnecessary if fs supports async/await
+function read(filePath) {
+  return new Promise(function(resolve, reject){
+    fs.readFile(filePath, {encoding: 'utf8'}, function(err, result){
+      if (err) { return reject(err); }
+      resolve(result.toString());
+    });
+  });
+}
+
+(async () => {
+  let files = await read(path.join(dir, 'index.txt'));
+  let data = await* files
+    .match(/^.*(?=\n)/gm)
+    .map((fileName) => path.join('input', fileName))
+    .map(read);
+
+  process.stdout.write(data.join(''));
+}());
```

Comment:
[Babel has a weird bug](https://babeljs.slack.com/archives/general/p1438027027001390) with `import` and `require` statement for the polyfill and generator functions in global context so I had to wrap it in an IIFE. Will fiddle with it more to see if I can make it work.